src/applications/MachineShopSimulator.java

```diff
@@ -47,38 +47,37 @@ boolean moveToNextMachine(Job theJob, SimulationResults simulationResults) {
      * 
      * @return last job run on this machine
      */
-    Job changeState(int theMachine) {// Task on theMachine has finished,
+    private Job changeState(int theMachine) {// Task on theMachine has finished,
                                             // schedule next one.
         Job lastJob;
-        if (machine[theMachine].getActiveJob() == null) {// in idle or change-over
+        if (machine[theMachine].activeJob == null) {// in idle or change-over
                                                     // state
             lastJob = null;
             // wait over, ready for new job
-            if (machine[theMachine].getJobQ().isEmpty()) // no waiting job
+            if (machine[theMachine].jobQ.isEmpty()) // no waiting job
                 eList.setFinishTime(theMachine, largeTime);
             else {// take job off the queue and work on it
-                machine[theMachine].setActiveJob((Job) machine[theMachine].getJobQ()
-                        .remove());
-                machine[theMachine].setTotalWait(machine[theMachine].getTotalWait() + timeNow
-                        - machine[theMachine].getActiveJob().getArrivalTime());
-                machine[theMachine].setNumTasks(machine[theMachine].getNumTasks() + 1);
-                int t = machine[theMachine].getActiveJob().removeNextTask();
+                machine[theMachine].activeJob = ((Job) machine[theMachine].jobQ.remove());
+                machine[theMachine].totalWait =
+                        (machine[theMachine].totalWait + timeNow - machine[theMachine].activeJob.getArrivalTime());
+                machine[theMachine].numTasks++;
+                int t = machine[theMachine].activeJob.removeNextTask();

```

Comment:
This chunk of code is doing a lot of work I think would be better handled in Machine. 
I think the "data class" smell may have been reduced, but I think now it only stores data and is kind of a lazy class which is worse imo.
I feel like to resolve the data class you went the wrong direction by pulling stuff into MSS, when it might have been better to pull logic into Machine.

Reply:
I think this comment is spot on. I think you want to be moving logic _into_ `Machine`, not pulling it out of it.